src/main/java/codes/biscuit/skyblockaddons/core/SkyblockKeyBinding.java

```diff
@@ -63,16 +63,24 @@ public String getKeyName() {
      */
     public boolean isKeyDown() {
         int keyCode = this.getKeyCode();
+        
+        if (!registered || keyCode == 9999) {
+            return false;
+        }
 
-        if (registered && keyCode != 9999) {
-            if (keyCode < 0) {
-                return Mouse.isButtonDown(keyCode + 100);
-            } else {
-                return Keyboard.isKeyDown(keyCode);
+        if (keyCode < 0) {
+            int button = keyCode + 100;
+            if (button < 0 || button >= Mouse.getButtonCount()) {
```

Comment:
This condition looks like unnecessary because of:
https://github.com/LWJGL/lwjgl/blob/2df01dd762e20ca0871edb75daf670ccacc89b60/src/java/org/lwjgl/input/Mouse.java#L398-L401